### PR TITLE
pin managed clusters' K8s version on stable branches

### PR DESCRIPTION
**Pull request by nbusseneau**
Please review per commit.

Resources:

- AKS
  - Run `az aks get-versions --location westeurope --output table`
  - https://learn.microsoft.com/en-us/azure/aks/supported-kubernetes-versions

- EKS
  - Run `eksctl version -o json | jq -r '.EKSServerSupportedVersions[]'`
    (note: this asssumes `eksctl` is up to date, as this is static info)
  - https://docs.aws.amazon.com/eks/latest/userguide/kubernetes-versions.html
  - https://eksctl.io/usage/schema/#metadata-version

- GKE
  - Run `gcloud container get-server-config --format="yaml(channels)"`
  - https://cloud.google.com/kubernetes-engine/versioning
  - https://cloud.google.com/kubernetes-engine/docs/release-schedule

### CODE REVIEW

**Comment by nbusseneau** [comment]
Further discussion:

- How do we ensure stable branches that need to be updated over time (e.g. 1.13 here) keep in sync? 
- What do we do about `master`?
  - Do we also pin and then bump as cloud providers provide new versions?
  - If we don't, probably we should try to have `master` use preview / rapid channels to ensure that it tests new K8s version as early as possible. I would advocate for this solution, though it has the same disadvantage, which is that CI might suddenly break due to cloud provider updates, but I think that's useful noise.

**Comment by nbusseneau** [comment]
We have split the commit pinning 1.10 workflows to a separate PR #22734 due to issues in testing, so that we can proceed with at least unblocking the more recent stable branches. Discussion points above are still relevant though (as they're not specific to 1.10).

**Comment by nbusseneau** [comment]
Actually the reason 1.10 testing has failed is because it's been done against `master` and not the 1.10 stable branch. I will reintegrate 1.10 changes in this PR and create separate PR for testing each of the stable branches.

**Comment by nbusseneau** [comment]
1.10: OK except for AKS (see discussion above)
- AKS: https://github.com/cilium/cilium/actions/runs/3697029585 failing due to K8s 1.21 not being available anymore on AKS.
- AWS-CNI: https://github.com/cilium/cilium/actions/runs/3697029582
- EKS: https://github.com/cilium/cilium/actions/runs/3697029593
- GKE: https://github.com/cilium/cilium/actions/runs/3705492260/jobs/6279474808
- External workloads: https://github.com/cilium/cilium/actions/runs/3705492235/jobs/6279473688
- Multicluster: https://github.com/cilium/cilium/actions/runs/3697029589

1.11: OK
- AKS: https://github.com/cilium/cilium/actions/runs/3697072807 (failure due to flakes but bootstrap OK)
- AWS-CNI: https://github.com/cilium/cilium/actions/runs/3697072804
- EKS: https://github.com/cilium/cilium/actions/runs/3697072811
- GKE: https://github.com/cilium/cilium/actions/runs/3697072791
- External workloads: https://github.com/cilium/cilium/actions/runs/3697072790
- Multicluster: https://github.com/cilium/cilium/actions/runs/3697072799

1.12: OK
- AKS: https://github.com/cilium/cilium/actions/runs/3697094458
- AWS-CNI: https://github.com/cilium/cilium/actions/runs/3697094431
- EKS: https://github.com/cilium/cilium/actions/runs/3697094437
- GKE: https://github.com/cilium/cilium/actions/runs/3697094444
- External workloads: https://github.com/cilium/cilium/actions/runs/3697094439
- Multicluster: https://github.com/cilium/cilium/actions/runs/3697094446

1.13: OK
- AKS: https://github.com/cilium/cilium/actions/runs/3697121897
- AWS-CNI: https://github.com/cilium/cilium/actions/runs/3697121894
- EKS: https://github.com/cilium/cilium/actions/runs/3697121893
- GKE: https://github.com/cilium/cilium/actions/runs/3697121911 (failure due to flakes but bootstrap OK)
- External workloads: https://github.com/cilium/cilium/actions/runs/3697121901
- Multicluster: https://github.com/cilium/cilium/actions/runs/3697121915

Master: OK
- AKS: https://github.com/cilium/cilium/actions/runs/3697151172
- AWS-CNI: https://github.com/cilium/cilium/actions/runs/3697151168
- EKS: https://github.com/cilium/cilium/actions/runs/3697151185
- GKE: https://github.com/cilium/cilium/actions/runs/3697151169
- External workloads: https://github.com/cilium/cilium/actions/runs/3697151175
- Multicluster: https://github.com/cilium/cilium/actions/runs/3697151164

File: .github/workflows/conformance-aks-v1.10.yaml [hunk 1]
```diff
@@ -66,6 +66,7 @@ env:
   name: ${{ github.repository_owner }}-${{ github.event.repository.name }}-${{ github.run_id }}
   location: westeurope
   cost_reduction: --node-vm-size Standard_B2s --node-osdisk-size 30
+  k8s_version: 1.21
```
Comment (nbusseneau) [inline .github/workflows/conformance-aks-v1.10.yaml]: We need to decide what to do for AKS on 1.10, cf. discussion above.

Comment (nbusseneau) [reply .github/workflows/conformance-aks-v1.10.yaml]: Actually, in order to merge this ASAP, I'll just use K8s 1.22 for AKS on 1.10, and we can decide on how to answer the other points later.

**Review by gandro — APPROVED** [review]

**Review by tklauser — APPROVED** [review]
:heart: the clean commit history and the additional info in the commit messages.

**Review by tommyp1ckles — APPROVED** [review]

**Review by pchaigno — APPROVED** [review]
Nice commit descriptions!

**Comment by nbusseneau** [comment]
Reviews are in and CI passed (see links to runs above), marking as ready to merge.

**Pull request merged**